### ISSUE TITLE
keycloak: password authentication failed for user "bn_keycloak" at cluster reboot 

### ISSUE THREAD

**Issue by marmila** [ricsanfre/pi-cluster#256]
Hello!
Just to track an issue that I am experiencing with the 1.8.0 version of the project: 

### **Name and Version**
bitnami/keycloak 17.3.6

### **What steps will reproduce the issue?**
I have install the chart without any special parameters as declared in the repo; the application will bootstrap normally. To reproduce the issue, it is sufficient to perform a graceful shutdown of the cluster (make shutdown-k3s-worker/master/gateway).

### **What is the expected behavior?**
keycloak should be able to connect to the created database and startup.

### **What do you see instead?**
The keycloak-0 pod crashes because it is not able to connect to the database.
The log of keycloak-postgresql-0 shows:

FATAL:  password authentication failed for user "bn_keycloak"
DETAIL:  Connection matched pg_hba.conf line 1: "host     all             all

A similar issue is tracked [here](https://github.com/bitnami/charts/issues/9795) 
As reported by @ricsanfre , _The solution should be similar to the one I applied for oauth2-proxy redis password:  define Posgresql admin password [using a external secret](https://picluster.ricsanfre.com/docs/sso/#alternative-installation-using-external-secret-gitops-1) instead of letting the helm chart to generate a random one._

At the moment, as workaround I have implemented a cleanup script on crontab to remove the pvc used by keycloak at reboot. 
Thanks!

**Comment by ghelobytes** [comment]
I wonder if explicitly setting `postgresql.auth.password`  in `values.yaml` will mitigate this issue. 
- https://artifacthub.io/packages/helm/bitnami/keycloak/17.3.6
<img width="304" alt="image" src="https://github.com/ricsanfre/pi-cluster/assets/1610267/e28afefe-9185-4643-a0ce-946d67d1a350">

**Comment by ricsanfre** [comment]
Thanks @marmila for opening the issue.

As @ghelobytes is commenting, thanks for your contribution, I am sure this issue might be solved setting all passwords and not letting the helm chart create random passwords.

With this particular keycloak helm chart, random passwords have issues when upgrading the helm charts if existing passwords are not provided during upgrade.See details in [bitnami's keycloak helm chart documentation: How to manage passwords](https://docs.bitnami.com/kubernetes/apps/keycloak/configuration/manage-passwords/)

I am going to change the way keycloak is deployed so posgresql credentials can be specified within an external secrets.

```yaml
  # postgresSQL
  postgresql:
    enabled: true
    auth:
      username: keycloak
      database: keycloak
      existingSecret: keycloak-secret
      secretKeys:
        adminPasswordKey: postgresql-admin-password
        userPasswordKey: password
    architecture: standalone
```

**Comment by ricsanfre** [comment]
I have merged PR https://github.com/ricsanfre/pi-cluster/pull/262 into master branch.

Please @marmila can you try new keycloak packaged application and see if this issue is solved?. 

Notice that new credentials have been added to Vault, so you will need to add them manually or reset the full cluster configuration and start fresh.
The new required vault entries are: 
- postgresql/admin
- keycloak/postgresql 

Regards
Ricardo

**Comment by marmila** [comment]
Hello everyone!

@ricsanfre : today I have merged the changes from PR #262  in my repo, then started fresh after a cluster reset. 
I can confirm that the issue, after a graceful cluster shutdown,  is no longer present, and keycloak starts with no errors.
Many many thanks for the support and have a nice day!
